R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+    return()
+  
+  legendTable <- createJaspTable("Hypothesis Legend")
+  deps <- switch(type, 
+                 "regression" = c("model", "covariates"),
+                 "anova" = c("model", "fixedFactors"),
+                 "ancova" = c("model", "fixedFactors"),
+                 "sem" = c("model", "syntax"))
+  legendTable$dependOn(options = deps)
+  legendTable$position <- position
+  legendTable$addColumnInfo(name = "number",     type = "string", title = "")
+  legendTable$addColumnInfo(name = "hypothesis", type = "string", title = gettext("Hypothesis"))
+  jaspResults[["legendTable"]] <- legendTable
+  
+  if (options[["model"]] != "") {
+    rest.string <- .bainCleanModelInput(options[["model"]])
+    hyp.vector <- unlist(strsplit(rest.string, "[;]")) 
+    for (i in 1:length(hyp.vector)) {
+      row <- list(number = gettextf("H%i", i), hypothesis = hyp.vector[i])
+      legendTable$addRows(row)
+    }
+  } else {
+    if(type == "regression"){
+      variables <- options[["covariates"]]
+      if (length(variables) == 0) {
+        row <- list(number = gettext("H1"), hypothesis = "")
+      } else if (length(variables) == 1) {
+        row <- list(number = gettext("H1"), hypothesis = paste(variables, "= 0"))
+      } else {
+        row <- list(number = gettext("H1"), hypothesis = paste0(paste0(variables, " = 0"), collapse = " & "))
+      }
+    } else if(type == "anova" || type == "ancova"){
+      if (options[["fixedFactors"]] != "") {
+        string <- paste(paste(options[["fixedFactors"]], levels(dataset[, options[["fixedFactors"]]]), sep = ""), collapse = " = ")
+        row <- list(number = gettext("H1"), hypothesis = string)
+      } else {
+        row <- list(number = gettext("H1"), hypothesis = "")
+      }
+    } else if(type == "sem"){
+      row <- list(number = gettext("H1"), hypothesis = "")
+    }
+    legendTable$addRows(row)
+  }
+}
+
+# Create a table containing the main analysis results
+.bainTestResultsTable <- function(dataset, options, bainContainer, missing, ready, type, position) {
+  
+  if (!is.null(bainContainer[["mainResultsTable"]])) 
+    return()
+  
+  title <- switch(type,
+                  "independentTTest" = gettext("Bain Independent Samples Welch's T-Test"),
+                  "pairedTTest" = gettext("Bain Paired Samples T-Test"),
+                  "onesampleTTest" = gettext("Bain One Sample T-test"),
+                  "anova" = gettext("Bain ANOVA"),
+                  "ancova" = gettext("Bain ANCOVA"),
+                  "regression" = gettext("Bain Linear Regression"),
+                  "sem" = gettext("Bain Structural Equation Model"))
+  deps <- switch(type,
+                 "independentTTest" = c("variables", "bayesFactorType", "hypothesis"),
+                 "pairedTTest" = c("pairs", "hypothesis", "bayesFactorType"),
+                 "onesampleTTest" = c("variables", "hypothesis", "bayesFactorType"),
+                 "anova" = NULL,
```

Comment:
Are the deps set on the container or something? Seems odd that they are empty

Reply:
The dependencies on the total analysis (all results) are set on a container within each analysis' R file. These dependencies are solely on the main table (shared function), but they are empty for ANOVA for example because when you change any option there all results get refreshed. However, for regression for example, if you click standardized then only the main results and coefficients table are changed, but the descriptions table can stay.